### DISCUSSION TITLE
36. CORS의 시나리오를 설명해주세요. 어떤 식으로 매커니즘이 작동하나요?

### DISCUSSION THREAD

**Discussion by linirini** [jewelry-box/cs-interview#37] [General]
.

**Comment by jongmee** [comment]
## 단순 요청
사전 요청(Preflight Request)을 보내지 않는 요청으로 [특정 조건들](https://developer.mozilla.org/ko/docs/Web/HTTP/CORS#%EB%8B%A8%EC%88%9C_%EC%9A%94%EC%B2%ADsimple_requests)을 모두 만족시켜야 한다. 따로 CORS 헤더를 검사하지 않기 때문에 서버 측에서 CSRF 대비를 따로 해야 한다.

HTML4.0의 form element를 통한 요청이 대표적이다.

## 비단순 요청
전 요청(Preflighted requests)으로 서버의 정책을 확인한다. OPTIONS 메서드로 요청하면 Access-Control-Allow-Origin, Access-Control-Allow-Methods, Access-Control-Allow-Headers 응답 헤더로 확인한다.
